### PR TITLE
Stop using `Refined`

### PR DESCRIPTION
**Pull request by houcros**
I remember we discussed getting rid of the `Refined` library, and that's why we implemented our `ValidatedGeneric` in libats. (Unfortunately I don't remember exactly why, but @koshelev might be able to fill that gap.) So, if this is still something worth doing, we can gradually do this for the various services and eventually remove `Refined` as a dependency in libats.

To be merged and deployed with https://github.com/advancedtelematic/ota-plus-server/pull/1997.

### CODE REVIEW

File: src/main/scala/com/advancedtelematic/ota/deviceregistry/DevicesResource.scala [hunk 1]
```diff
@@ -153,8 +150,8 @@ class DevicesResource(
     complete(db.run(InstalledPackages.getDevicesCount(pkg, ns)))
 
   def listPackagesOnDevice(device: DeviceId): Route =
-    parameters(('regex.as[String Refined Regex].?, 'offset.as[Long].?, 'limit.as[Long].?)) { (regex, offset, limit) =>
```
Comment (houcros) [inline src/main/scala/com/advancedtelematic/ota/deviceregistry/DevicesResource.scala]: I'm waiting for FE confirmation, but it seems no one uses this regex parameter and we can just remove it.

Comment (houcros) [reply src/main/scala/com/advancedtelematic/ota/deviceregistry/DevicesResource.scala]: Ok so it is used, but regex is overkill. I will replace for "string contains".

Comment (houcros) [reply src/main/scala/com/advancedtelematic/ota/deviceregistry/DevicesResource.scala]: Done.

Comment (simao) [reply src/main/scala/com/advancedtelematic/ota/deviceregistry/DevicesResource.scala]: but we need to change ui?

Comment (houcros) [reply src/main/scala/com/advancedtelematic/ota/deviceregistry/DevicesResource.scala]: Yes, I'll make sure to sync with the FE before merging this.

Comment (houcros) [reply src/main/scala/com/advancedtelematic/ota/deviceregistry/DevicesResource.scala]: Created parallel PR in app: https://github.com/advancedtelematic/ota-plus-server/pull/1997.

**Comment by houcros** [comment]
Because it's a transitive dependency that we get from libats. The idea would be to stop using this in whatever services use it and then eventually remove it from libats.

**Comment by zabbal** [comment]
> Because it's a transitive dependency that we get from libats. 

I'm still puzzled. If we have "import eu.timepit.refined.api.Refined" in our .scala than it looks like an explicit dependency. What am I missing?

**Comment by houcros** [comment]
Yes, but if you see there is no declaration of the library in the `build.sbt` in this repo. There is, nonetheless, a declaration to include libats, and that in turn includes `refined` as a dependency here: https://github.com/advancedtelematic/libats/blob/cc903491991c2376fe840d421db5fe82593fec8e/build.sbt#L68.

**Pull request merged**